certstore_linux.go

```diff
@@ -1,14 +1,321 @@
 package certstore
 
-import "errors"
+/*
+#cgo pkg-config: nss
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <stdlib.h>
 
-// This will hopefully give a compiler error that will hint at the fact that
-// this package isn't designed to work on Linux.
-func init() {
-	CERTSTORE_DOESNT_WORK_ON_LINIX
+#include <nss.h>
+#include <cryptohi.h>
+#include <nspr.h>
+#include <p12.h>
+#include <p12plcy.h>
+#include <pk11pub.h>
+#include <prerror.h>
+#include <secerr.h>
+
+SECItem *P12U_NicknameCollisionCallback(SECItem *old_nick, PRBool *cancel, void *wincx) {
+	char *nick = NULL;
+	SECItem *ret_nick = NULL;
+	CERTCertificate *cert = (CERTCertificate *)wincx;
+	if (!cancel || !cert) {
+		return NULL;
+	}
+	nick = CERT_MakeCANickname(cert);
+	if (!nick) {
+		return NULL;
+	}
+	if (old_nick && old_nick->data && old_nick->len &&
+		PORT_Strlen(nick) == old_nick->len - 1 &&
+		!PORT_Strncmp((char *)old_nick->data, nick, old_nick->len - 1)) {
```

Comment:
I'm just realizing that this whole function was copied from NSS ([source](https://github.com/nss-dev/nss/blob/master/cmd/pk12util/pk12util.c#L123-L170)). There, they seem to assume that `old_nick->len` doesn't account for a trailing null byte. Have you managed to test this? I'm wondering if this is a bug in NSS or if we're misunderstanding how strings in `SECItem`s work.

Reply:
I was a bit unsure about the legality of copy-pasting from the NSS repo. I checked with GitHub's legal team and they say this is dangerous. This function should only come into play when trying to import something with a non-unique nickname, right? It seems like we could get by without the logic in this function for now. Could we change this function to just `return NULL` for now?

Reply:
I wasn't really sure how much this function was needed either.. I'll have a ponder